### ISSUE TITLE
Improvements to Mochawesome Report Generation and Publishing

### ISSUE THREAD

**Issue by CBonade** [department-of-veterans-affairs/va.gov-team#28265]
With the MVP for this feature functioning, some improvements have been suggested that might smooth this process out even more.  The following steps should be researched into and/or implemented:

- [x] The `testing-tools` repository should auto-delete any reports or videos that are over 30 days old.
- [x] Move the Cypress Tests Summary to the end of the workflow and include the link to the Mochawesome report as part of the summary.
- [x] Modify the Cypress Tests Summary to include data from all runs compiled, similar to how the Mochawesome report currently does.  Currently the Cypress Tests Summary only shows data from the most recently completed parallel process.
- [x] Look into removing our dependency on a third party action for the Cypress Tests Summary and consider generating our own summary.

After extensively exploring the github repository route for this, S3 has been decided on as a more stable, more functional solution. We've had a bucket created for this and while S3 access has been granted, access to the bucket for this is still restricted so we are just awaiting some sorting issues. Ideally, the process will be essentially the same: Push the report and videos to S3, link the S3 hosted report back to the workflow.  The only difference, is we will have a 30 day object expiration timer set in S3 to handle the clutter.

**Comment by CBonade** [comment]
The base has been set for  uploading to S3 has been set today.  While I've copied the authentication as it's used in other areas where uploading is being done, it appears our credentials that are used in other parts of the workflow do not have access to upload to the new bucket that was created (though, I do now have access to view the bucket).  WIP.

**Comment by CBonade** [comment]
Today I inquired with ops about what was needed to make this work, and a ticket requesting the proper access has been filed.  Awaiting further movement from ops on this.

**Comment by CBonade** [comment]
Still awaiting S3 access for the front end service account to our testing tools reports bucket. Issue is open in vagov.team and tagged for ops

**Comment by CBonade** [comment]
Access was given, however it has incorrect permissions. Continuing to work through this with ops.

**Comment by CBonade** [comment]
Happy to report that reports are now uploading to S3 as part of the standard workflow.  The only things I am waiting on at this point are additional S3 permissions to delete and set object expiration timers

**Comment by CBonade** [comment]
Moving this to Review/QA to be verified for review for closing as the primary portion of the task has been completed.  Cleaned up, followup task created here: https://app.zenhub.com/workspaces/vsp-5cedc9cce6e3335dc5a49fc4/issues/department-of-veterans-affairs/va.gov-team/29609

**Comment by pjhill** [comment]
Excellent, thanks for sticking to this!
